indispensable-josef/src/commonMain/kotlin/at/asitplus/signum/indispensable/josef/JweDecrypted.kt

```diff
@@ -7,25 +7,25 @@ package at.asitplus.signum.indispensable.josef
  *
  * @see [JweEncrypted]
  */
-data class JweDecrypted(
+data class JweDecrypted<T: Any>(
     val header: JweHeader,
-    val payload: ByteArray
+    val payload: T,
 ) {
     override fun equals(other: Any?): Boolean {
         if (this === other) return true
         if (other == null || this::class != other::class) return false
 
-        other as JweDecrypted
+        other as JweDecrypted<*>
 
         if (header != other.header) return false
-        if (!payload.contentEquals(other.payload)) return false
+        if (!payload.equals(other.payload)) return false
 
         return true
     }
 
     override fun hashCode(): Int {
         var result = header.hashCode()
-        result = 31 * result + payload.contentHashCode()
+        result = 31 * result + payload.hashCode()
```

Comment:
this needs discrimination: hashcode for primitive arrays (sadly, each one separately), Array<*> and regular equals.